packages/shared/src/graphql/posts.ts

```diff
@@ -444,8 +448,38 @@ export const SUBMIT_EXTERNAL_LINK_MUTATION = gql`
   }
 `;
 
-interface SubmitExternalLink {
+export interface ExternalLinkPreview {
+  title: string;
+  image: string;
+}
+
+export interface ExternalLink extends Partial<ExternalLinkPreview> {
   url: string;
+}
+
+export const PREVIEW_LINK_MUTATION = gql`
+  mutation CheckLinkPreview($id: ID!) {
+    checkLinkPreview(id: $id) {
+      title
+      image
+    }
+  }
+`;
+
+interface ExternalLinkPreviewProps {
+  url: string;
+  sourceId: string;
+}
+
+export const getExternalLinkPreview = async (
+  params: ExternalLinkPreviewProps,
+): Promise<ExternalLinkPreview> => {
+  const res = await request(graphqlUrl, PREVIEW_LINK_MUTATION, params);
+
+  return res.checkLinkPreview;
```

Comment:
What happens if this doesn't exist?

Reply:
Should throw an error and the process would continue without preview.